tests/distributions/test_gaussian_mixtures.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import absolute_import, division, print_function
+
+import math
+
+import torch
+
+import pytest
+from pyro.distributions import MixtureOfDiagNormalsSharedCovariance
+from tests.common import assert_equal
+
+
+@pytest.mark.parametrize('mix_dist', [MixtureOfDiagNormalsSharedCovariance])
+@pytest.mark.parametrize('K', [3])
+@pytest.mark.parametrize('D', [2, 4])
+@pytest.mark.parametrize('batch_mode', [True, False])
+@pytest.mark.parametrize('flat_logits', [True, False])
+@pytest.mark.parametrize('cost_function', ['quadratic'])
+def test_mean_gradient(K, D, flat_logits, cost_function, mix_dist, batch_mode):
+    n_samples = 2000 * 1000
+    if batch_mode:
+        sample_shape = torch.Size(())
+    else:
+        sample_shape = torch.Size((n_samples,))
+    locs = torch.tensor(torch.rand(K, D), requires_grad=True)
+    sigmas = torch.ones(D) + 0.5 * torch.rand(D)
+    sigmas = torch.tensor(sigmas, requires_grad=True)
+    if not flat_logits:
+        logits = torch.tensor(1.5 * torch.rand(K), requires_grad=True)
+    else:
+        logits = torch.tensor(0.1 * torch.rand(K), requires_grad=True)
+    omega = torch.tensor(0.2 * torch.ones(D) + 0.1 * torch.rand(D), requires_grad=False)
+
+    _pis = torch.exp(logits)
+    pis = _pis / _pis.sum()
+
+    if cost_function == 'cosine':
```

Comment:
Either remove this if statement or add `cosine` to the `@pytest.mark.parametrize` decorator

Reply:
why can't i keep an unused test around so that it's available during future refactoring? it's only a few lines of code

Reply:
If it's not tested it will break.

Reply:
it is tested. with another cost function.